python/ray/tests/test_kill_raylet_signal_log.py

```diff
@@ -1,28 +1,51 @@
-import psutil
+import os
 import signal
 import sys
+import time
 import pytest
+import threading
+
 import ray
 
+from ray.test_utils import (
+    wait_for_pid_to_exit, )
+
+SIGKILL = signal.SIGKILL if sys.platform != "win32" else signal.SIGTERM
+
+
+def check_result(path):
+    with open(path) as f:
+        time.sleep(1)
+        s = f.read()
+        assert len(s) > 0
+        print(s)
+        assert "SIGTERM" in s
 
-def get_pid(name):
-    pids = psutil.process_iter()
-    for pid in pids:
-        if (pid.name() == name):
-            return pid.pid
+
+class check_thread(threading.Thread):
+    def __init__(self, raylet_out_path):
+        threading.Thread.__init__(self)
+        self._daemonic = True
```

Comment:
Why set this?

Reply:
The _daemonic make the thread detach.
If we use thread.join() here, the log of SIGTERM can not be flushed, use detach thread can avoid the problem.

Reply:
But in this case, the test will pass without waiting for the thread to finish. How do you ensure the signal is in the log file?